it/controllers/benefits/medical/MedicalOrDentalBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,583 @@
+      "Redirect to cya page when all is valid in new data flow" which {
+        //should change to childcare later
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true,
+            Some(benefits(fullMedicalChildcareEducationModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(url(taxYearEOY), form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+
+      }
+    }
+    "redirect to another page when the request is valid but they aren't allowed to view the page and" should {
+
+      val user = UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedAgentEN))
+
+      lazy val form: Map[String, String] = Map(AmountForm.amount -> "97")
+
+
+      "Redirect user to the check your benefits page with no cya data in session" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(url(taxYearEOY), form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+      }
+
+      "Redirect user to the tax overview page when in year" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(
+            fullBenefits.map(_.copy(benefits = fullBenefits.flatMap(_.benefits.map(_.copy(telephone = None))))))))), nino, taxYearEOY)
+          insertCyaData(employmentUserData(isPrior = false, cyaModel("name", hmrc = true,
+            Some(benefits(fullMedicalChildcareEducationModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(url(taxYear), form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+        }
+      }
+
+      "redirect to the other services question page when there is a medical insurance amount but the other medicalInsuranceQuestion has not been answered" +
```

Comment:
again rename this to make it more easily understandable of whats going on in the test